shared/events.js

```diff
@@ -280,6 +281,13 @@ export class GroupContract extends HashableContract {
       },
       AcknowledgeFounder (state, data) {
         if (!state.members.find(username => username === data.username)) { state.members.push(data.username) }
+      },
+      // TODO: remove group profile when leave group is implemnted
+      ProfileAdjustment (state, data) {
+        let profile = state.profiles[data.username]
+        if (!profile) { profile = state.profiles[data.username] = {} }
+        if (!data.value) { return delete profile[data.name] }
+        profile[data.name] = data.value
```

Comment:
Confused.

- what is `data.username`?
- what is `data.name` and how is it different?
- what is `data.value`?
- what is `state.profiles`?

And how can you use any of the above data fields when they aren't defined in `ProfileAdjustment.fields`?